File: LATIHAN4DPBO2023/Python/Course.py
from Dosen import Dosen
# inisialisasi class Course
class Course:
	__namaMatkul = ""
	__dosenAmpu = Dosen()
	__mhsKontrak = []

	# constructor
	# untuk dosenAmpu dan mhsKontrak akan berisi object, composite dari class Dosen dan class Mahasiswa
	def __init__(self, dosen, mahasiswa, namaMatkul = ""):
		self.__mhsKontrak = []
		self.__namaMatkul = namaMatkul
		self.__dosenAmpu = dosen
		# karena list maka diinputnya menggunakan append
		self.__mhsKontrak.append(mahasiswa)

	# kumpulan method setter dan getter untuk tiap atribut
	def setNamaMatkul(self, namaMatkul):
		self.__namaMatkul = namaMatkul

	def getNamaMatkul(self):
		return self.__namaMatkul

	def setDosenAmpu(self, dosen):
		self.__dosenAmpu = dosen

	def getDosenAmpu(self):
		return self.__dosenAmpu

	def setMhsKontrak(self, mahasiswa):
		self.__mhsKontrak.append(mahasiswa)

	def getMhsKontrak(self):
		return self.__mhsKontrak

File: LATIHAN4DPBO2023/CPP/Course.cpp
/*Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 4
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin.*/

//include file header
#include "header.hh"

//inisialisasi constructor
Course::Course()
{
	namaMatkul = "";
}

//inisialisasi constructor dengan parameter
Course::Course(string namaMatkul, Dosen dosen, Mahasiswa mahasiswa)
{
	this->namaMatkul = namaMatkul;
	this->mahasiswa.push_back(mahasiswa);
	this->dosen = dosen;
}

//inisialisasi method setter dan getter
void Course::setNamaMatkul(string namaMatkul)
{
	this->namaMatkul = namaMatkul;
}

string Course::getNamaMatkul()
{
	return this->namaMatkul;
}

void Course::setMahasiswa(Mahasiswa mahasiswa)
{
	this->mahasiswa.push_back(mahasiswa);
}

list<Mahasiswa> Course::getMahasiswa()
{
	return this->mahasiswa;
}

void Course::setDosen(Dosen dosen)
{
	this->dosen = dosen;
}

Dosen Course::getDosen()
{
	return this->dosen;
}

//inisialisasi destructor
Course::~Course()
{
}